src/main/java/edu/harvard/h2ms/seeders/UserSeeder.java

```diff
@@ -47,19 +50,25 @@ public void seed(ContextRefreshedEvent event) {
   private void seedUserTable() {
 
     if (userRepository.count() == 0) {
-      List<List<String>> records = asList(asList("Default", "User", "admin@h2ms.org"));
+      List<List<String>> records = asList(asList("Default", "User", "admin@h2ms.org", "ROLE_ADMIN"),
+              asList("Ben", "Jenkins", "bjenkins@h2ms.org", "ROLE_OBSERVER"),
+              asList("Wesley", "Skillerns", "wskillern@h2ms.org", "ROLE_USER"));

```

Comment:
May not want to hard code specific users on user seeder.  (As stephen mentioned before) Seeders are for populating minimum dataset for production server if it's not already defined (in this case only admin is needed).

Reply:
@shin @stbenjam
I had requested Michael seed some users with each role so Wesley will be able to test his changes. Is there anyway these can be seeded for our testing site only?

Reply:
Why not just create them? Don't we have a UI for users? Seeds should be usable in a fresh installation, and kept clean.  Someone deploying the app other than us has no need for a Ben Jenkins user. 

You can have a conditional seeder by adding a `@Profile` annotation to it, and we could have demo data.  We talked about do it a while back but I don't think I got a card to implement it and other things took priority before Milestone 2.  We can revisit that if you like.

Reply:
We don't have a UI for roles yet. And when we do I'd be concerned about how tedious the process will be. Since we will have to register 2 new accounts and reset their passwords every time we wanted to test anything that relies on roles.

You are right, in practice a default seeded observer/user probably wouldn't be useful for everyone. They really just care about the admin.

If you can do it for our own testing I'd appreciate it. Otherwise, Wesley can just try mocking around not having users with other roles in the database.

Reply:
There's really no need to do it repeatedly through the UI or go through the registration process. Postman can save requests, or you could write a Ruby/Pthon script. All you need to do is POST this to the users end point (https://test.h2ms.org/api/users).

Or see if Michael has time to address your concern here, or create a card and assign it to someone.

```json
{
  "email": "bjenkins@h2ms.org",
  "enabled": true,
  "firstName": "Ben",
  "lastName": "Jenkins",
  "password": "password",
  "roles": [
      "http://test.h2ms.org/api/roles/11"
  ],
  "type": "string",
  "verified": true
}
```

Reply:
TBH you might find it more flexible to have a script that seeds data for a demo, so you don't have to submit a pull request every time to update a seeder.

Reply:
So you actually use that password. I was under the impression that the backend ignored it and generated a random password. That is why Kevin thought it wasn't required.

If you actually store the password so I don't have to go through that process then that is fine.

Side note, this is not for a demo. This is for testing. I don't care about having a "Ben Jenkins" user, I would be cool with having a "Default User", "Default Admin", and "Default Observer" just so the frontend can be sure everything is locked down appropriately for all 3.

Reply:
He might have been confused since it looks like a random token is generated and set as the password.
![](https://imgur.com/K0s6b0D)